### PR TITLE
Refetch validating keys if no keys are fetched

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**

When a validator starts with a remote wallet, the wallet can contain no keys. In such case the validator is stuck indefinitely on `WaitForActivation`, and adding keys to the wallet does not unblock it.

**Which issues(s) does this PR fix?**

N/A

**Other notes for review**

When redistributing keys by scaling up validator replicas, the safest way is to scale up validators first, and then assign keys to them. This will not work without the fix, as the new validator replicas will not fetch their keys.

### CODE REVIEW

File: validator/client/validator.go [hunk 1]
```diff
@@ -236,8 +245,34 @@ func (v *validator) WaitForActivation(ctx context.Context) error {
...
+				}
+				if len(keys) == 0 {
+					log.Warn(msgNoKeysFetched)
+					continue
+				}
+				validatingKeys = keys
+				break
+			case <-ctx.Done():
+				log.Debug("Context closed, exiting fetching validating keys")
+				return errors.New("context closed, no longer attempting to fetch validating keys")
```
Comment (terencechain) [inline validator/client/validator.go:270]:
```suggestion
				return ctx.Err()
```

I like `ctx.Err()` better here. Current error message sounds duplicated with the debug log

**Review by terencechain — APPROVED** [review]

File: validator/client/validator.go [hunk 2]
```diff
@@ -238,8 +247,33 @@ func (v *validator) WaitForActivation(ctx context.Context) error {
...
+			case <-ticker.C:
+				keys, err := v.keyManager.FetchValidatingPublicKeys(ctx)
+				if err != nil {
+					return errors.Wrap(err, msgCouldNotFetchKeys)
+				}
+				if len(keys) == 0 {
+					log.Warn(msgNoKeysFetched)
+					continue
+				}
+				validatingKeys = keys
```
Comment (prestonvanloon) [inline validator/client/validator.go:268]: Should this return after or do we need to continuously fetch validating public keys?

Comment (rkapka) [reply validator/client/validator.go:268]: After this statement we jump out of `select` and immediately hit `break`, which will jump out of the `for` loop. I had a `break` statement here initially, but DeepSource complained that it's redundant.

Comment (prestonvanloon) [reply validator/client/validator.go]: Got it. Thanks for the comment!

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**